src/main/scala/com/cognite/sdk/scala/common/login.scala

```diff
@@ -12,14 +14,23 @@ class Login[F[_]](val requestSession: RequestSession[F]) {
   implicit val loginStatusDecoder = deriveDecoder[LoginStatus]
   implicit val dataLoginStatusDecoder = deriveDecoder[DataLoginStatus]
   def status(): F[LoginStatus] =
-    requestSession
-      .sendCdf { request =>
-        request
-          .get(uri"${requestSession.baseUri}/login/status")
-          .response(asJson[DataLoginStatus])
-          .mapResponse {
-            case Left(value) => throw value.error
-            case Right(value) => value.data
-          }
-      }
+    try {
+      requestSession
+        .sendCdf { request =>
+          request
+            .get(uri"${requestSession.baseUri}/login/status")
+            .response(asJson[DataLoginStatus])
+            .mapResponse {
+              case Left(value) => throw value.error
+              case Right(value) => value.data
+            }
+        }
+    } catch {
+      case unknownHost: UnknownHostException => throw unknownHost
+      case connectionTimeout: ConnectException => throw connectionTimeout
+      case _: Throwable =>
+        throw new RuntimeException(
+          "Could not connect to Cognite API. Make sure the URI correctly points to your cognitedata tenant."
```

Comment:
We should only mention "the URI" if it has been changed from the default. Otherwise people will wonder about the URI even if they're just outside of WiFi range. As a compromise we could say "Could not connect to Cognite Data Fusion API using baseUri ${requestSession.baseUri}`.